lib/ansible/module_utils/common/process.py

```diff
@@ -6,19 +6,27 @@
 import os
 
 from ansible.module_utils.common.file import is_executable
+from ansible.module_utils.common.warnings import deprecate
 
 
 def get_bin_path(arg, opt_dirs=None, required=None):
```

Comment:
Is it worth changing the default to a sentinel value so we also warn when people do `required=None`? Might be overkill here but just thought I should bring it up.